llvm/lib/MC/MCMachOStreamer.cpp

```diff
@@ -232,7 +232,7 @@ void MCMachOStreamer::emitAssemblerFlag(MCAssemblerFlag Flag) {
 }
 
 void MCMachOStreamer::emitLinkerOptions(ArrayRef<std::string> Options) {
-  getWriter().getLinkerOptions().push_back(Options);
```

Comment:
I am a bit worried about all other `getWriter()` instance in this file. They basically do not work with MCCAS.

Reply:
To do that, we would have to move data members`std::optional<unsigned> PtrAuthABIVersion;` and `MCLOHContainer LOHContainer;` etc to `MCObjectWriter` which seems like a layering violation to me, these are MachO specific things that `MCObjectWriter` shouldn't know about. What do you think?

Reply:
The alternative is subclass MachOCASWriter under MachOObjectWriter. I don't know how much overhead is needed for that.

Reply:
Can we just instant crash on `getWriter()` and disable all tests that hit that crash? I can see we might not really emit the other fields but we don't want UB in the code.